packages/components/src/button/test/index.tsx

```diff
@@ -254,10 +254,10 @@ describe( 'Button', () => {
 			);
 
 			expect(
-				screen.getByRole( 'button', {
-					description: 'Description text',
+				screen.getByRole( 'tooltip', {
+					hidden: true,
 				} )
-			).toBeVisible();
+			).toHaveTextContent( 'Description text' );
```

Comment:
I'd prefer to keep the existing assertion for the button, since it tests the existence of the actual button described by  the visually hidden description text. Then I'd just add the assertion you created here for the tooltip, since that tests the hidden tooltip part. 

In other words, this tests two different parts of what's rendered and to me it makes sense to test them both for a better test coverage.

Reply:
Fair point!

Reply:
Just flagging that, when #54312 merged, the tooltip won't be found in the DOM until visible (ie. triggered with hover/focus)